docs/content/python/getting-started/application/model.md

```diff
@@ -0,0 +1,214 @@
+code. The Oso library loads and evaluates policy files when your
+application runs. Now that you've installed Oso, create a policy file
+called `main.polar` and use it in your app:
+
+{{< literalInclude path="examples/add-to-your-application/python/app/oso.py" >}}
+
+To setup Oso in your app, you must...
+
+- initialize `Oso`: Usually Oso will be initialized globally and used
+during every request to enforce authorization.
+- tell Oso what data types you will authorize: With Oso, you express
+  authorization logic over the data types used in your application. Often
+  these will be model classes, but they could be any type of data or
+  resource you want to control access to.
+- load your policy: Oso policies are written in policy files. Call
+  `load_files` with the paths to your policy files.
+
+## Define your resources, permissions and roles
+
+Authorization controls access to *resources* in your application. In
+this guide, we're going to show how to implement the most common
+authorization model, role-based access control, with Oso. See [How to:
+modeling](TODO) **TODO** to see how to implement other authorization
+models in Oso.
+
+To start with, you need to define resource blocks in your policy. Let's
+say we have an application like GitHub that includes a `Repository`
+resource. Define the `Repository` resource in `main.polar`:
+
+```polar
+actor User {}
+
+resource Repository {
+    permissions = ["read", "push"];
+    roles = ["contributor", "maintainer"];
+
+    # A user has the "read" permission if they have the
+    # "contributor" role.
+    "read" if "contributor";
+
+    # A user has the "push" permission if they have the
+    # "maintainer" role.
+    "push" if "maintainer";
+}
+```
+
+This policy declares `"read"` and `"push"` as permissions for the
+`Repository` resource and assigns them to roles. We also tell Oso
+that our `User` class is an `actor` with the `actor User {}`
+declaration.
+
+The `"read" if "contributor";` statement is an example of a *shorthand rule.*
+Add an `"admin"` role and give it its own permissions by adding some
+more shorthand rules:
+
+```polar
+resource Repository {
+    permissions = ["read", "push", "delete"];
+    roles = ["contributor", "maintainer", "admin"];
+
+    "read" if "contributor";
+    "push" if "maintainer";
+
+	# A user has the "delete" permission if they have the
+	# "admin" role.
+	"delete" if "admin";
+
+	# A user has the "maintainer" role if they have
+	# the "admin" role.
+    "maintainer" if "admin";
+
+	# A user has the "contributor" role if they have
+	# the "maintainer" role.
+    "contributor" if "maintainer";
+}
+```
+
+The last rules we added are between two roles: A user has the
+`"maintainer"` role and all permissions associated with it if they have the
+`"admin"` role, and the same for `"contributor"` and `"maintainer"`.
+
+### Giving your users roles
+
+Now that we've finished our policy, we must associate users with roles
```

Comment:
Sounds like the policy is "finished" here but not "set up" until the last section which is a little unclear. But I'm not sure what would be a good way to make it clear we're talking about the rules in the resource blocks. "Now that we've written the core of our policy" or something of that kind?